src/Control/MonadZero.purs

```diff
@@ -29,28 +39,4 @@ import Data.Unit (Unit, unit)
 -- | laws:
 -- |
 -- | - Annihilation: `empty >>= f = empty`
-class (Monad m, Alternative m) <= MonadZero m
-
-instance monadZeroArray :: MonadZero Array
-
--- | Fail using `Plus` if a condition does not hold, or
--- | succeed using `Monad` if it does.
--- |
--- | For example:
--- |
--- | ```purescript
--- | import Prelude
--- | import Control.Monad (bind)
--- | import Control.MonadZero (guard)
--- | import Data.Array ((..))
--- |
--- | factors :: Int -> Array Int
--- | factors n = do
--- |   a <- 1..n
--- |   b <- 1..n
--- |   guard $ a * b == n
--- |   pure a
--- | ```
-guard :: forall m. MonadZero m => Boolean -> m Unit
-guard true = pure unit
-guard false = empty
+class (Monad m, Alternative m, Warn (Text "'MonadZero' is deprecated")) <= MonadZero m
```

Comment:
On reflection isn’t this message a bit too dry? Shouldn’t we also give a suggestion?

I’m thinking of something like:

```
'MonadZero' is deprecated, use 'Monad' and 'Alternative' constraints instead
```

Reply:
Yes, I'm very much in favor of suggestions when there's a clear replacement we can recommend.